signingscript/src/signingscript/sign.py

```diff
@@ -927,6 +931,44 @@ def make_signing_req(input_file, fmt, keyid=None, extension_id=None):
     return sign_req
 
 
+@time_function
+def make_files_signing_req(input_files, fmt, keyid=None, options=None):
+    """Make a signing request to pass to the /sign/files endpoint in autograph."""
+    signing_req = {"keyid": keyid, "options": options}
+    encoded_signing_req = BytesIO()
+    encoded_signing_req.write(b"[{")
+    for k, v in signing_req.items():
+        encoded_signing_req.write(json.dumps(k).encode("utf8"))
+        encoded_signing_req.write(b":")
+        encoded_signing_req.write(json.dumps(v).encode("utf8"))
+        encoded_signing_req.write(b",")
```

Comment:
Why are we building a JSON object string by string here? Can't we build up `signing_req` as a dictionary and then return `json.dumps(signing_req).encode("utf-8")` at the end?

Reply:
I am pretty sure this is equivalent. Building the byte string this way is one of the things I tried when I couldn't figure out why I was getting 400 from autograph (I think it was an issue with "." in the file names after all)

[it's close to what write_signing_req_to_disk is doing](https://github.com/mozilla-releng/scriptworker-scripts/blob/master/signingscript/src/signingscript/sign.py#L814)... I can simplify it

Reply:
Ah, that makes sense.. I guess we sign some pretty big blobs and autograph needs the whole contents of the file in the request. By building it piece by piece we don't have the whole file loaded in memory at once.

Reply:
I made this func return a dict instead of the byte string to make it more orthogonal with `make_signing_req`